test/integration/metadataAPITests.coffee

```diff
@@ -0,0 +1,327 @@
+        testMetadata = {}
+        testMetadata = JSON.parse JSON.stringify sampleMetadata
+        testMetadata.Channels = [{
+            name: "TestChannel2"
+            allow: [ "PoC", "Test1", "Test2" ]
+            routes: [{ name: "test route", host: "localhost", port: 9876, primary: true }]
+            txViewAcl: "group1"
+        }]
+        request("https://localhost:8080")
+          .post("/metadata/validate")
+          .set("auth-username", testUtils.rootUser.email)
+          .set("auth-ts", authDetails.authTS)
+          .set("auth-salt", authDetails.authSalt)
+          .set("auth-token", authDetails.authToken)
+          .send(testMetadata)
+          .expect(201)
+          .end (err, res) ->
+            if err
+              done err
+            else
+              res.body.errors.length.should.equal 1
+              res.body.successes.length.should.equal 4
+              done()
+                
+      it  "should fail to validate invalid metadata and return status 400 - bad request", (done) ->
+        testMetadata = {}
+        testMetadata = JSON.parse JSON.stringify sampleMetadata
+        testMetadata.InvalidField = { "name": "value" }
+        request("https://localhost:8080")
+          .post("/metadata/validate")
+          .set("auth-username", testUtils.rootUser.email)
+          .set("auth-ts", authDetails.authTS)
+          .set("auth-salt", authDetails.authSalt)
+          .set("auth-token", authDetails.authToken)
+          .send(testMetadata)
+          .expect(400)
+          .end (err, res) ->
+            if err
+              done err
+            else
+              done()
+      
+      it  "should not allow a non admin user to validate metadata", (done) ->
+        request("https://localhost:8080")
+          .post("/metadata/validate")
+          .set("auth-username", testUtils.nonRootUser.email)
+          .set("auth-ts", authDetails.authTS)
+          .set("auth-salt", authDetails.authSalt)
+          .set("auth-token", authDetails.authToken)
+          .send(sampleMetadata)
+          .expect(403)
+          .end (err, res) ->
+            if err
+              done err
+            else
+              done()  
+      
+      it  "should return 404 if not found", (done) ->
+        request("https://localhost:8080")
+          .post("/metadata/validate/bleh")
+          .set("auth-username", testUtils.rootUser.email)
+          .set("auth-ts", authDetails.authTS)
+          .set("auth-salt", authDetails.authSalt)
+          .set("auth-token", authDetails.authToken)
+          .send(sampleMetadata)
+          .expect(404)
+          .end (err, res) ->
+            if err
+              done err
+            else
+              done()         
+                
+                
+                
+                
+                
+                
+                
+                
```

Comment:
`removeProperties` and `getUniqueIdentifierForCollection` should be unit tested in their own unit test file as well. Those are good examples of testable functions.